pages/requests/[uuid].js

```diff
@@ -38,10 +39,29 @@ const Request = () => {
   const { allSOWs, isLoadingSOWs, isSOWError } = useAllSOWs(uuid, request?.identifier, session?.accessToken)
   const { messages, isLoadingMessages, isMessagesError, mutateMessages, messagesData } = useMessages(uuid, session?.accessToken)
   const { files, isLoadingFiles, isFilesError, mutateFiles, filesData } = useFiles(uuid, session?.accessToken)
-  const documents = (allSOWs) ? [...allSOWs] : []
 
-  const isLoading = isLoadingRequest || isLoadingSOWs || isLoadingFiles || isLoadingMessages
+  const [allPOs, setAllPOs] = useState(false)
+  const [isPOError, setIsPOError] = useState(false)
+  const [isPOLoading, setIsPOLoading] = useState(false)
+  useEffect(() => {
+    const getPOsAsync = async () => {
+      const response = await getAllPOs(request?.quotedWareID, uuid, request?.identifier, session?.accessToken)
+      if (response.allPOs) {
+        setAllPOs(response.allPOs)
+      } else if (response.isPOError) {
+        setIsPOError(response.isPOError)
+      } else {
+        setIsPOLoading(true)
+      }
+    }
+    getPOsAsync()
+  }, [allPOs, isPOError, request?.quotedWareID, uuid, request?.identifier, session?.accessToken])
+  
+  console.log(allPOs)
+  const isLoading = isLoadingRequest || isLoadingSOWs || isLoadingFiles || isLoadingMessages || isPOLoading
   const isError = isRequestError || isSOWError || isFilesError|| isMessagesError
```

Comment:
when isPOError is added here, it will cause the page to get stuck in error state since before the response from the `getAllPOs` finishes, an error returns. 

instead isPOError is mainly used as a dependency of the useEffect, to check that the response changes and makes the useEffect get the value of allPOs when it is finally ready

There may be a better way to set up the error/loading part but for now this catches any errors before they are thrown in the user facing component, and it just takes an extra second for the PO to show up due to the useEffect not being called until after the page finishes rendering